<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GrupoEstudoDialog</class>
 <widget class="QDialog" name="GrupoEstudoDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>üìö Grupos de Estudo</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
/* ====== FUNDO PRINCIPAL COM IMAGEM ====== */
QDialog#GrupoEstudoDialog {
    background-image: url(:/images/imagens/fundo_2.png);
    background-position: center;
    background-repeat: no-repeat;
}

/* ====== BARRA DE NAVEGA√á√ÉO ====== */
QLabel#navBar {
    background-color: #0f1011;
}

/* ====== OVERLAY SEMI-TRANSPARENTE ====== */
QWidget#centralWidget {
    background-color: rgba(26, 22, 26, 0.85);
    border-radius: 15px;
}

/* ====== T√çTULO PRINCIPAL ====== */
QLabel#tituloLabel {
    color: #F4B315;
    font-size: 28px;
    font-weight: bold;
    padding: 15px;
    background-color: transparent;
}

/* ====== SUBT√çTULO ====== */
QLabel#subtituloLabel {
    color: #8E6915;
    font-size: 14px;
    background-color: transparent;
}

/* ====== LOGO E √çCONES ====== */
QLabel#logoLabel, QLabel#homeButton, QLabel#perfilButton {
    background-color: transparent;
    border-radius: 50px;
}

/* ====== BARRA DE BUSCA ====== */
QLineEdit#searchLine {
    background-color: rgb(236, 236, 236);
    border-radius: 8px;
    padding: 8px;
    font-size: 13px;
}

/* ====== BOT√ïES PRINCIPAIS ====== */
QPushButton {
    background-color: #F4B315;
    color: #1A161A;
    border: none;
    border-radius: 8px;
    padding: 12px 24px;
    font-weight: bold;
    font-size: 13px;
}

QPushButton:hover {
    background-color: #D3AF35;
}

QPushButton:pressed {
    background-color: #8E6915;
}

/* ====== BOT√ïES DE NAVEGA√á√ÉO ====== */
QPushButton#materiasButton, QPushButton#gruposButton, QPushButton#criarButton {
    background-color: #423738;
    color: #F4B315;
    border: 2px solid #F4B315;
}

QPushButton#materiasButton:hover, QPushButton#gruposButton:hover, QPushButton#criarButton:hover {
    background-color: #524447;
    border-color: #D3AF35;
}

/* ====== BOT√ÉO FECHAR ====== */
QPushButton#fecharButton {
    background-color: #423738;
    color: #F4B315;
    border: 2px solid #F4B315;
}

QPushButton#fecharButton:hover {
    background-color: #524447;
    border-color: #D3AF35;
}

/* ====== COMBOBOX (FILTRO) ====== */
QComboBox {
    background-color: rgba(66, 55, 56, 0.9);
    color: #F4B315;
    border: 2px solid #F4B315;
    border-radius: 6px;
    padding: 8px 12px;
    font-size: 13px;
    font-weight: bold;
}

QComboBox:hover {
    border-color: #D3AF35;
    background-color: rgba(82, 68, 71, 0.9);
}

QComboBox::drop-down {
    border: none;
    width: 30px;
}

QComboBox::down-arrow {
    image: none;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-top: 6px solid #F4B315;
    margin-right: 8px;
}

QComboBox QAbstractItemView {
    background-color: #423738;
    color: #F4B315;
    selection-background-color: #F4B315;
    selection-color: #1A161A;
    border: 2px solid #F4B315;
}

/* ====== LINEEDIT ====== */
QLineEdit {
    background-color: #423738;
    color: #F4B315;
    border: 2px solid #8E6915;
    border-radius: 8px;
    padding: 10px;
    font-size: 14px;
}

QLineEdit:focus {
    border-color: #F4B315;
}

/* ====== SPINBOX ====== */
QSpinBox {
    background-color: #423738;
    color: #F4B315;
    border: 2px solid #8E6915;
    border-radius: 8px;
    padding: 10px;
    font-size: 14px;
}

QSpinBox:focus {
    border-color: #F4B315;
}

/* ====== RADIO BUTTONS ====== */
QRadioButton {
    color: #F4B315;
    font-size: 14px;
    background-color: transparent;
}

QRadioButton::indicator {
    width: 20px;
    height: 20px;
    border-radius: 10px;
    border: 2px solid #F4B315;
    background-color: #423738;
}

QRadioButton::indicator:checked {
    background-color: #F4B315;
    border: 2px solid #F4B315;
}

/* ====== SCROLL AREA ====== */
QScrollArea {
    border: 2px solid #423738;
    border-radius: 10px;
    background-color: rgba(26, 22, 26, 0.7);
}

QScrollBar:vertical {
    background-color: #423738;
    width: 12px;
    border-radius: 6px;
}

QScrollBar::handle:vertical {
    background-color: #F4B315;
    border-radius: 6px;
    min-height: 20px;
}

QScrollBar::handle:vertical:hover {
    background-color: #D3AF35;
}

/* ====== CONTAINER ====== */
QWidget#materiasContainer, QWidget#salasContainer {
    background-color: transparent;
}

/* ====== LINHA SEPARADORA ====== */
Line {
    background-color: #F4B315;
}

/* ====== LABELS GERAIS ====== */
QLabel {
    color: #F4B315;
    background-color: transparent;
}

QLabel#infoLabel {
    color: #8E6915;
    font-size: 12px;
    font-style: italic;
}
   </string>
  </property>
  
  <layout class="QVBoxLayout" name="mainLayoutOuter">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   
   <item>
    <!-- ============================================================ -->
    <!-- BARRA DE NAVEGA√á√ÉO SUPERIOR -->
    <!-- ============================================================ -->
    <widget class="QWidget" name="navWidget" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>80</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>80</height>
      </size>
     </property>
     
     <!-- Logo UTFPR -->
     <widget class="QLabel" name="logoLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>80</width>
        <height>80</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">border-radius: 20px;</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="icons.qrc">:/images/imagens/Logo_1.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     
     <!-- Barra de Busca -->
     <widget class="QLineEdit" name="searchLine">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>25</y>
        <width>900</width>
        <height>35</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(236, 236, 236);</string>
      </property>
      <property name="placeholderText">
       <string>üîç Buscar grupos...</string>
      </property>
     </widget>
     
     <!-- Bot√£o Home -->
     <widget class="QLabel" name="homeButton">
      <property name="geometry">
       <rect>
        <x>1030</x>
        <y>20</y>
        <width>45</width>
        <height>45</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
      <property name="styleSheet">
       <string notr="true">border-radius: 50px;</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="icons.qrc">:/images/imagens/inicio.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     
     <!-- Bot√£o Perfil -->
     <widget class="QLabel" name="perfilButton">
      <property name="geometry">
       <rect>
        <x>1090</x>
        <y>20</y>
        <width>45</width>
        <height>45</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
      <property name="styleSheet">
       <string notr="true">border-radius: 50px;</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="icons.qrc">:/images/imagens/perfil.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     
     <!-- Fundo da barra -->
     <widget class="QLabel" name="navBar">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1200</width>
        <height>80</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #0f1011;</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     
     <zorder>navBar</zorder>
     <zorder>logoLabel</zorder>
     <zorder>searchLine</zorder>
     <zorder>homeButton</zorder>
     <zorder>perfilButton</zorder>
    </widget>
   </item>
   
   <item>
    <!-- Widget central -->
    <widget class="QWidget" name="centralWidget">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <property name="leftMargin">
       <number>20</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>20</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      
      <!-- T√çTULO -->
      <item>
       <widget class="QLabel" name="tituloLabel">
        <property name="text">
         <string>üìö Grupos de Estudo EducaUTFPR</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      
      <!-- SUBT√çTULO -->
      <item>
       <widget class="QLabel" name="subtituloLabel">
        <property name="text">
         <string>Conecte-se com colegas e estude em grupo</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      
      <!-- LINHA SEPARADORA -->
      <item>
       <widget class="Line" name="line">
        <property name="styleSheet">
         <string notr="true">background-color: #F4B315;</string>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth">
         <number>2</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      
      <!-- BOT√ïES DE NAVEGA√á√ÉO -->
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>15</number>
        </property>
        <item>
         <widget class="QPushButton" name="materiasButton">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>üìñ Mat√©rias</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="gruposButton">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>üë• Grupos</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="criarButton">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>‚ûï Criar</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      
      <!-- STACKED WIDGET (3 TELAS) -->
      <item>
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        
        <!-- ========== P√ÅGINA 1: MAT√âRIAS ========== -->
        <widget class="QWidget" name="page_materias">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>üîé Filtrar por categoria:</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: #F4B315; font-size: 14px; font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="categoriaComboBox">
            <property name="minimumSize">
             <size>
              <width>300</width>
              <height>40</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>üìö Todas as Mat√©rias</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üìê Matem√°tica</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>‚öõÔ∏è F√≠sica</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üíª Programa√ß√£o</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üóÑÔ∏è Banco de Dados</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üåê Redes e Sistemas</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üîß Engenharia</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>‚öóÔ∏è Qu√≠mica</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>üå± Humanas e Sociais</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QScrollArea" name="scrollArea">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <widget class="QWidget" name="materiasContainer">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>1136</width>
               <height>450</height>
              </rect>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        
        <!-- ========== P√ÅGINA 2: GRUPOS ========== -->
        <widget class="QWidget" name="page_grupos">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>üîì Entrar em Grupo Privado (com c√≥digo):</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: #F4B315; font-size: 14px; font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLineEdit" name="codigoGrupoEdit">
              <property name="minimumHeight">
               <number>40</number>
              </property>
              <property name="placeholderText">
               <string>C√≥digo do grupo (ex: 12345)</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="senhaGrupoEdit">
              <property name="minimumHeight">
               <number>40</number>
              </property>
              <property name="placeholderText">
               <string>üîí Senha (se necess√°rio)</string>
              </property>
              <property name="echoMode">
               <enum>QLineEdit::Password</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="entrarGrupoButton">
              <property name="minimumSize">
               <size>
                <width>120</width>
                <height>40</height>
               </size>
              </property>
              <property name="text">
               <string>üö™ Entrar</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>üåç Grupos P√∫blicos Dispon√≠veis:</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: #F4B315; font-size: 14px; font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QScrollArea" name="scrollArea_2">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="salasContainer">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>1136</width>
               <height>400</height>
              </rect>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        
        <!-- ========== P√ÅGINA 3: CRIAR GRUPO ========== -->
        <widget class="QWidget" name="page_criar">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>‚ûï Criar Novo Grupo de Estudo</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: #F4B315; font-size: 18px; font-weight: bold;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>üìù Nome do Grupo:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="nomeSalaEdit">
            <property name="minimumHeight">
             <number>40</number>
            </property>
            <property name="placeholderText">
             <string>Ex: Grupo de C√°lculo 1 - Turma A</string>
            </property>
           </widget>
          </item>
          
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>üìö Mat√©ria:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="materiaComboBox">
            <property name="minimumHeight">
             <number>40</number>
            </property>
           </widget>
          </item>
          
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>üîì Tipo do Grupo:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="tipoPublicaRadio">
              <property name="text">
               <string>üåç P√∫blico (sem senha)</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="tipoPrivadaRadio">
              <property name="text">
               <string>üîí Privado (com senha)</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>üîê Senha (apenas para grupos privados):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="senhaEdit">
            <property name="minimumHeight">
             <number>40</number>
            </property>
            <property name="placeholderText">
             <string>Digite uma senha</string>
            </property>
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>üë• M√°ximo de Participantes:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="maxParticipantesSpinBox">
            <property name="minimumHeight">
             <number>40</number>
            </property>
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          
          <item>
           <widget class="QPushButton" name="confirmarCriarButton">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>‚úÖ Criar Grupo</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      
      <!-- RODAP√â COM INFO E BOT√ÉO FECHAR -->
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="infoLabel">
          <property name="text">
           <string>üí° Dica: Crie grupos privados para estudar com amigos espec√≠ficos</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #8E6915; font-size: 12px; font-style: italic;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="fecharButton">
          <property name="minimumSize">
           <size>
            <width>120</width>
            <height>45</height>
           </size>
          </property>
          <property name="text">
           <string>‚úï Fechar</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 
 <resources>
  <include location="icons.qrc"/>
 </resources>
 
 <connections>
  <connection>
   <sender>fecharButton</sender>
   <signal>clicked()</signal>
   <receiver>GrupoEstudoDialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1100</x>
     <y>770</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  
  <connection>
   <sender>tipoPrivadaRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>senhaEdit</receiver>
   <slot>setEnabled(bool)</slot>
  </connection>
 </connections>
</ui>